### ISSUE TITLE
Error when trying to connect to AppleTV

### ISSUE THREAD

**Issue by pluto202** [postlund/pyatv#1938]
### What do you need help with?

Hello all, I would need some help to get peat running. I did now the second approach with the same error. I startet with a clean installation on a RaspberryPi and now tried it on a fresh Debian in a Proxmox Container. I have a VLAN segmented network but installed the pyatv hosts in the same network. I have an iobroker yahaka adapter running in the same network and I'm able to connect to the Apple TV Home and can manage my devices so connectivity to the AppleTV is possible.

I installed the pyatv in the following way on both hosts:

sudo apt-get update
sudo apt-get upgrade
sudo apt install python3-pip
pip3 install pyatv
sudo mv ~/.local/bin/atvscript /usr/bin/atvscript
sudo mv ~/.local/bin/atvremote /usr/bin/atvremote

afterwards I started a scan and was able to see all my devices in my different networks:

atvscript scan

{"result": "success", "datetime": "2023-03-11T17:27:14.736185+01:00", "devices": [{"name": "denon6700", "address": "10.10.20.100", "identifier": "00:06:78:41:E4:D2", "device_info": {"model": "Unknown", "model_str": "Denon AVC-X6700H", "operating_system": "Unknown", "version": null}, "services": [{"protocol": "airplay", "port": 7000}, {"protocol": "raop", "port": 7000}]}, {"name": "Wohnzimmer (2)", "address": "10.10.20.10", "identifier": "6A114DA4-2FA1-4F75-B4B4-D0699DB63952", "device_info": {"model": "Gen4K", "model_str": "Apple TV 4K", "operating_system": "TvOS", "version": "16.3.2"}, "services": [{"protocol": "companion", "port": 49153}, {"protocol": "airplay", "port": 7000}, {"protocol": "mrp", "port": 49154}, {"protocol": "raop", "port": 7000}]}, {"name": "Kueche", "address": "10.10.50.106", "identifier": "83C20C7CE8AE", "device_info": {"model": "Unknown", "model_str": "ShairportSync", "operating_system": "Unknown", "version": null}, "services": [{"protocol": "raop", "port": 5000}]}, {"name": "Wohnzimmer", "address": "10.10.12.204", "identifier": "F4:34:F0:3A:4D:E6", "device_info": {"model": "HomePodMini", "model_str": "HomePod Mini", "operating_system": "TvOS", "version": "16.3.2"}, "services": [{"protocol": "airplay", "port": 7000}, {"protocol": "companion", "port": 49155}, {"protocol": "raop", "port": 7000}]}]}

afterwards I wanted to connect to the AppleTV and wanted to see what is running there but I got an error back:

atvscript -s 10.10.20.10 playing
{"result": "failure", "datetime": "2023-03-11T17:28:31.799163+01:00", "exception": "app is not supported", "stacktrace": "Traceback (most recent call last):\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/scripts/atvscript.py\", line 308, in appstart\n    print(args.output(await _handle_command(args, abort_sem, loop)), flush=True)\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/scripts/atvscript.py\", line 204, in _handle_command\n    return await _run_command(atv, args, abort_sem, loop)\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/scripts/atvscript.py\", line 211, in _run_command\n    return output_playing(await atv.metadata.playing(), atv.metadata.app)\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/support/shield.py\", line 72, in _guard_method\n    return func(self, *args, **kwargs)\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/core/facade.py\", line 239, in app\n    return self.relay(\"app\")\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/core/relayer.py\", line 90, in relay\n    instance = self._find_instance(\n  File \"/home/manager/.local/lib/python3.9/site-packages/pyatv/core/relayer.py\", line 114, in _find_instance\n    raise exceptions.NotSupportedError(f\"{target} is not supported\")\npyatv.exceptions.NotSupportedError: app is not supported\n"}

Also the approach to pair with the AppleTV failed

atvremote -s 10.10.20.10 --protocol mrp pair
2023-03-11 17:29:53 ERROR [pyatv.scripts.atvremote]: Pairing failed
Traceback (most recent call last):
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/protocol.py", line 274, in _receive
    await semaphore.acquire()
  File "/usr/lib/python3.9/asyncio/locks.py", line 413, in acquire
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/support/__init__.py", line 33, in error_handler
    return await func(*args, **kwargs)
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/auth.py", line 40, in start_pairing
    await self.protocol.start(skip_initial_messages=True)
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/protocol.py", line 142, in start
    self.device_info = await self.send_and_receive(
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/protocol.py", line 261, in send_and_receive
    return await self._receive(identifier, timeout)
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/protocol.py", line 274, in _receive
    await semaphore.acquire()
  File "/home/manager/.local/lib/python3.9/site-packages/async_timeout/__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "/home/manager/.local/lib/python3.9/site-packages/async_timeout/__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/scripts/atvremote.py", line 179, in pair
    await self._perform_pairing(pairing)
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/scripts/atvremote.py", line 189, in _perform_pairing
    await pairing.begin()
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/protocols/mrp/pairing.py", line 51, in begin
    return await error_handler(
  File "/home/manager/.local/lib/python3.9/site-packages/pyatv/support/__init__.py", line 35, in error_handler
    raise exceptions.ConnectionFailedError(str(ex)) from ex
pyatv.exceptions.ConnectionFailedError

Would be great if someone can help may what is missing in my approach.

Thanks all
Andreas

**Comment by postlund** [comment]
You are doing it correctly, but you should not pair MRP. All the other protocols should work as long as you have changed access settings to allow anyone on the same network. Then just provide the credentials you got from pairing when calling `atvscript`.

**Comment by pluto202** [comment]
Thanks a lot. I was able to pair via airplay now and can perform commands and get response. 

Just one other questions. To use MRP commands with Airplay credentials I used the following command:

atvremote -s 10.10.20.10 --protocol mrp --airplay-credentials 18cb3c9441a6e41c3082140d492674801f287dc869ab44667c6023f218098321:50904b99bc5d8308270e8cca91bb488821555d1bd00882fcbb3a60e98da33a5a:45363530304633422d413746342d344338412d384639302d383033384431343638454645:62626631303566382d393761362d346631352d386465372d613033356630373261616335 turn_off

is this right?
